projects/driver_controls/inc/ads1015.h

```diff
@@ -0,0 +1,55 @@
+#pragma once
+// Module for using ADS1015.
+// GPIO and Interrupt, I2C should be initiated.
+// The user also needs to create a struct ADS1015Storage which would persist accross functions.
+// Start the ads1015 using ads1015_init and then configure channels using ads1015_config_channel.
+// Read raw and converted values using ads1015_read_raw and ads1015_read_converted.
+#include <stdbool.h>
+#include "gpio.h"
+#include "i2c.h"
+#include "status.h"
+
+typedef enum {
+  ADS1015_ADDRESS_GND = 0,
+  ADS1015_ADDRESS_VDD,
+  ADS1015_ADDRESS_SDA,
+  ADS1015_ADDRESS_SCL,
+  NUM_ADS1015_ADDRESSES,
+} Ads1015Address;
+
+typedef enum {
+  ADS1015_CHANNEL_0 = 0,
+  ADS1015_CHANNEL_1,
+  ADS1015_CHANNEL_2,
+  ADS1015_CHANNEL_3,
+  NUM_ADS1015_CHANNELS,
+} Ads1015Channel;
+
+// The callback runs after each conversion from the channel.
+typedef void (*Ads1015Callback)(Ads1015Channel channel, void *context);
+
+typedef struct Ads1015Storage {
+  I2CPort i2c_port;
+  Ads1015Address i2c_addr;
+  GPIOAddress ready_pin;
+  int16_t channel_readings[NUM_ADS1015_CHANNELS];
+  Ads1015Channel current_channel;
+  uint8_t channel_enable;
```

Comment:
Fun thing to think about: [Struct Packing](http://webcache.googleusercontent.com/search?q=cache:1LzCTClc5s8J:www.catb.org/esr/structure-packing/+&cd=1&hl=en&ct=clnk&gl=ca) ([Wikipedia](https://en.wikipedia.org/wiki/Data_structure_alignment))

Reply:
I'd also recommend making it more obvious that this is a bitset.